commcare_connect/opportunity/utils/completed_work.py

```diff
@@ -35,3 +42,32 @@ def update_status(completed_works, opportunity_access, compute_payment=True):
     if compute_payment:
         opportunity_access.payment_accrued = payment_accrued
         opportunity_access.save()
+
+
+def update_work_payment_date(access: OpportunityAccess):
+    payments = Payment.objects.filter(opportunity_access=access).order_by("date_paid")
+    completed_works = CompletedWork.objects.filter(opportunity_access=access).order_by("status_modified_date")
+
+    paid = 0
+    works_to_update = []
+    completed_works_iter = iter(completed_works)
+
+    try:
+        current_work = next(completed_works_iter)
+    except StopIteration:
+        return
+
+    for payment in payments:
+        paid += payment.amount
+
+        while paid > current_work.payment_accrued:
```

Comment:
This is missing the logic deducts from the total paid when we assign that payment to a completed work. Right now this will run forever in most cases, because the total paid out is always greater than what is due for the next completed work. For example if each work accrues $5 and the first payment was $10, 10 is always greater than 5. We need to deduct what we are assigning to the work, or keep a similar running total for what we owe.